pkg/k8s/service_cache.go

```diff
@@ -871,11 +871,11 @@ func (s *ServiceCache) MergeClusterServiceDelete(service *serviceStore.ClusterSe
 
 	if ok {
 		s.emitEvent(ServiceEvent{
-			Action:    DeleteService,
-			ID:        id,
-			Service:   svc,
-			Endpoints: endpoints,
-			SWGDone:   swg.Add(),
+			Action:       DeleteService,
+			ID:           id,
+			OldService:   svc,
+			OldEndpoints: endpoints,
```

Comment:
Ah thanks for catching it. I somehow glossed over this one even though I checked all refs to Service field. 
Hmm, the OldService isn't set in mergeExternalServiceDeleteLocked which seems wrong. I'll set it, but would appreciate if you double-check that this won't cause surprises (shouldn't, essentially same as EnsureService).

Yep I already checked policy/k8s/service.go. There it's really just triggering the recalculation based on whether a CNP matches or not and it's not using the data passed in the event to actually perform the calculation.

Reply:
> Hmm, the OldService isn't set in mergeExternalServiceDeleteLocked which seems wrong. I'll set it, but would appreciate if you double-check that this won't cause surprises (shouldn't, essentially same as EnsureService).

Yep, it seems that `mergeServiceUpdateLocked` is potentially setting it to nil as well in certain cases. In terms of service handling it shouldn't matter at all, because `addK8sSVCs` actually treats `svc==oldSvc` and `oldSvc==nil` in the same way (the former just implies slightly more work). Differently, the policy piece would lead to a bit more work when `oldSvc==nil`, as it treats it as a new service. That said, I think that your change makes sense (it could make sense to adapt `mergeServiceUpdateLocked` as well to default it to `svc` when nil, just to ensure consistency). 

I'm definitely happy that this logic is going to be replaced in the future, because there are currently too many caveats and small differences that tracking all corner cases is super difficult right now.

> Yep I already checked policy/k8s/service.go. There it's really just triggering the recalculation based on whether a CNP matches or not and it's not using the data passed in the event to actually perform the calculation.

Perfect, that matches my understanding as well.